quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/ProxyPropertyMappers.java

```diff
@@ -37,9 +40,25 @@ public static PropertyMapper<?>[] getProxyPropertyMappers() {
                         .to("quarkus.http.proxy.allow-x-forwarded")
                         .mapFrom("proxy-headers")
                         .transformer((v, c) -> proxyEnabled(ProxyOptions.Headers.xforwarded, v, c))
+                        .build(),
+                fromOption(ProxyOptions.PROXY_TRUSTED_ADDRESSES)
+                        .to("quarkus.http.proxy.trusted-proxies")
+                        .validator((mapper, value) -> mapper.validateExpectedValues(value,
+                                (c, v) -> validateAddress(v)))
+                        .appendValidateEnabled(() -> !Configuration.isBlank(ProxyOptions.PROXY_HEADERS), "proxy-headers is set")
+                        .paramLabel("trusted proxies")
                         .build()
         };
     }
+    
+    private static void validateAddress(String address) {
+        if (Inet.parseCidrAddress(address) != null) {
+            return;
+        }
+        if (Inet.parseInetAddress(address) == null) {
+            throw new PropertyException(address + " is not a valid IPv4 nor IPv6 IP address");
+        }
+    }
```

Comment:
Do we need this? Quarkus doesn't perform any validations?

Reply:
Quarkus does perform validation, but based upon the issues raised and the general recommendation to not use hostnames we're restricting to a subset of what quarkus allows.

Reply:
It's then a question whether this is rather a Quarkus issue and as such should be raised with them. But I already added a comment about that above, so resolving this one.